microprot/scripts/calculate_Neff.py

```diff
@@ -0,0 +1,117 @@
+import re
+import math
+import click
+from scipy.cluster.hierarchy import linkage, fcluster
+from skbio import io, Protein, TabularMSA, DistanceMatrix
+from skbio.sequence.distance import hamming
+
+
+"""
+Calculate effective family size of a multiple sequence alignment with a given
+percent identity cutoff.
+"""
+
+
+def parse_msa_file(infile):
+    """Read sequences from a multiple sequence alignment (MSA) file.
+
+    Parameters
+    ----------
+    infile : str
+        file path to input MSA file in A3M format (like FASTA format, but
+        lowercase letters will be dropped)
+
+    Returns
+    -------
+    skbio TabularMSA
+    """
+    seqs = []
+    for seq in io.read(infile, format='fasta'):
+        seqs.append(Protein(re.sub('[a-z]', '', str(seq)),
+                            metadata=seq.metadata))
+    return TabularMSA(seqs)
+
+
+def hamming_distance_matrix(msa):
+    """Compute Hamming distance matrix of an MSA.
+
+    Parameters
+    ----------
+    msa : skbio TabularMSA
+        Aligned sequences for calculating pairwise Hamming distances
+
+    Returns
+    -------
+    skbio DistanceMatrix
+    """
+    return DistanceMatrix.from_iterable(msa, hamming, key='id', validate=False)
+
+
+def cluster_sequences(dm, cutoff):
+    """Perform hierarchical clustering based on a distance matrix.
+
+    Parameters
+    ----------
+    dm : skbio DistanceMatrix
+        aligned sequences for calculating pairwise Hamming distances
+    cutoff : float
+        sequence percent identity cutoff for defining clusters
+
+    Returns
+    -------
+    list of int
+        flat cluster numbers to which sequences are assigned
+    """
+    t = 1.0 - cutoff / 100.0
+    return list(fcluster(linkage(dm.condensed_form()), t,
+                         criterion='distance'))
+
+
+def effective_family_size(clusters, length):
```

Comment:
Add a flag to return either `Neff` as defined before, or `Nclu`, not scaled by the length of the sequence. Make it return `Neff` by default.